core/amber/src/main/scala/edu/uci/ics/amber/engine/common/statetransition/StateManager.scala

```diff
@@ -0,0 +1,71 @@
+package edu.uci.ics.amber.engine.common.statetransition
+
+import edu.uci.ics.amber.engine.common.statetransition.StateManager.{
+  IntermediateState,
+  InvalidStateException,
+  InvalidTransitionException
+}
+
+import scala.collection.mutable
+
+object StateManager {
+  case class InvalidStateException(message: String)
+      extends RuntimeException(message)
```

Comment:
Let's extend from WorkflowRuntimeException here. It's easier to always use only string. But I feel including the stacktrace is helpful. Therefore, creating a workflowRuntimeError is good. To get the current stacktrace, let's use `Thread.currentThread().getStackTrace().mkstring`